lib/pmdtester/builders/diff_report_builder.rb

```diff
@@ -0,0 +1,220 @@
+require 'nokogiri'
+require_relative './html_report_builder'
+
+module PmdTester
+  # Building diff report according to the output of DiffBuilder
+  class DiffReportBuilder < HtmlReportBuilder
+    NO_DIFFERENCES_MESSAGE = 'No differences found!'.freeze
+
+    def build(project, report_diff)
+      project.diffs_exist = report_diff.diffs_exist?
```

Comment:
This looks a bit strange: Is this the correct place, to handover this information?
Maybe it's better done, before the diffreportbuilder is called?

Reply:
Ok, I've now seen, where it is used. First, for each project, the html report is created - and this sets the flag "diffs_exist" on the project. The flag is later on used for the summary report. So it is crucial, to render the reports of each project into html before rendering the summary... this is a flow dependency.

How about using the project object for storing the results (the report_diff object), too? That way, the summary builder can ask the project, whether a diff exists and the project in turn asks the report diff.
The project object would start out with the basic information from the project list (the configuration with which PmdTester has been started) and we would add more information with the progress.
If you rework this, I suggest to do this in a new PR.